core/trino-main/src/test/java/io/trino/operator/aggregation/AbstractTestAggregationFunction.java

```diff
@@ -67,6 +70,58 @@ protected Object getExpectedValueIncludingNulls(int start, int length, int lengt
         return getExpectedValue(start, length);
     }
 
+    protected Block[] getSequenceBlocksForDoubleNaNTest(int start, int length)
+    {
+        BlockBuilder blockBuilder = DOUBLE.createBlockBuilder(null, length);
+        for (int i = start; i < start + length - 5; i++) {
+            DOUBLE.writeDouble(blockBuilder, i);
+        }
+        DOUBLE.writeDouble(blockBuilder, Double.NaN);
+        for (int i = start + length - 4; i < start + length; i++) {
+            DOUBLE.writeDouble(blockBuilder, i);
+        }
+        return new Block[] {blockBuilder.build()};
+    }
+
+    protected Block[] getSequenceBlocksForDoubleInfinityTest(int start, int length)
+    {
+        BlockBuilder blockBuilder = DOUBLE.createBlockBuilder(null, length);
+        for (int i = start; i < start + length - 5; i++) {
+            DOUBLE.writeDouble(blockBuilder, i);
+        }
+        DOUBLE.writeDouble(blockBuilder, Double.POSITIVE_INFINITY);
+        for (int i = start + length - 4; i < start + length; i++) {
+            DOUBLE.writeDouble(blockBuilder, i);
+        }
+        return new Block[] {blockBuilder.build()};
+    }
+
+    protected Block[] getSequenceBlocksForRealNaNTest(int start, int length)
```

Comment:
Let's move these utilities into their usage sites where we know that we're dealing with `REAL` vs `DOUBLE`, etc

Reply:
Done